test-data/unit/check-classes.test

```diff
@@ -2759,3 +2759,50 @@ class B(A):
 class C(B):
     x = ''
 [out]
+
+[case testInvalidMetaclassStructure]
+class X(type): pass
+class Y(type): pass
+class A(metaclass=X): pass
+class B(A, metaclass=Y): pass  # E: Inconsistent metaclass structure for 'B'
+
+[case testMetaclassNoTypeReveal]
+class M:
+    x = 0  # type: int
+
+class A(metaclass=M): pass
+
+reveal_type(A.x)  # E: Revealed type is 'builtins.int'
+
+[case testMetaclassTypeReveal]
+from typing import Type
+class M(type):
+    x = 0  # type: int
+
+class A(metaclass=M): pass
+
+reveal_type(A.x)  # E: Revealed type is 'builtins.int'
```

Comment:
I have removed it in the [last commit](https://github.com/python/mypy/pull/2475/commits/b083a9748f7b01564e8142abc734c87989e68bd2) but it still appears here.